File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        result = 0

        character_Set = set() # Python's in-built DS

        for right in range(len(s)):
            # This loop-checker needs to be here to avoid the infinite loop error
            # Now since set is full, we need to start removing the elements from the character_Set
            while s[right] in character_Set:
                character_Set.remove(s[left])
                left += 1
                
            # start adding elements in the set and then destroying them
            # after encountering duplicates
            character_Set.add(s[right])
            result = max(result, right - left + 1)

            
        return result

File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
   public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        vector<vector<int>> result;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > 0) break;  // Means there will be no duplicates

            if (i > 0 && nums[i] == nums[i - 1]) continue;

            int left = i + 1;
            int right = nums.size() - 1;

            while (left < right) {
                int ThreeSum = nums[i] + nums[left] + nums[right];

                if (ThreeSum < 0) {
                    left++;
                } else if (ThreeSum > 0) {
                    right--;
                } else {
                    result.push_back({nums[i], nums[left], nums[right]});
                    left++;
                    right--;

                    while (left < right && nums[left] == nums[left - 1]) {
                        left++;
                    }
                }
            }
        }
        return result;
    }
};
